readthedocs/search/tests/test_views.py

```diff
@@ -79,38 +82,121 @@ def _get_search_result(self, url, client, search_params):
+        return results, page
+
+    def _get_result_headings(self, page):
+        headings_anchor = page.find('.module-list-wrapper .search-result-item p a')
+        headings_text = headings_anchor.text()
+        return headings_text
 
-    @pytest.mark.parametrize('data_type', ['content', 'headers', 'title'])
+    @pytest.mark.parametrize('data_type', DATA_TYPES_VALUES)
     @pytest.mark.parametrize('page_num', [0, 1])
     def test_file_search(self, client, project, data_type, page_num):
-        query = get_search_query_from_project_file(project_slug=project.slug, page_num=page_num,
-                                                   data_type=data_type)
+        query = get_search_query_from_project_file(
+            project_slug=project.slug,
+            page_num=page_num,
+            data_type=data_type
+        )
 
-        result, _ = self._get_search_result(url=self.url, client=client,
-                                            search_params={'q': query, 'type': 'file'})
-        assert len(result) == 1
-        assert query in result.text()
+        results, _ = self._get_search_result(
+            url=self.url,
+            client=client,
+            search_params={ 'q': query, 'type': 'file' }
+        )
+
+        assert len(results) >= 1
+        for res in results:
+            fragments = res.cssselect('.fragment')
+            assert len(fragments) >= 1
+
+        assert query in results.text()
+
+    def test_file_search_show_correct_role_name(self, client):
+        """Test that searching files should show all role_names."""
+
+        # searching for '/api/v3/' to test that
+        # correct role_name are displayed
+        results, page = self._get_search_result(
+            url=self.url,
+            client=client,
+            search_params={ 'q': '/api/v3/', 'type': 'file' }
+        )
+
+        assert len(results) >= 1
+        for res in results:
+            fragments = res.cssselect('.fragment')
+            assert len(fragments) >= 1
+
+        content = page.find('.navigable .language-list')
+        # these are taken from files `support` and `wiping`
+        expected_role_names = ['http:get', 'http:patch', 'http:post']
+
+        assert len(content) == 3
+        for role_name in expected_role_names:
+            assert role_name in content.text()
+
+    def test_file_search_filter_role_name(self, client):
+        """Test that searching files filtered according to role_names."""
+
+        search_params = {
+            'q': 'notfound',
+            'type': 'file',
+        }
+
+        # searching without the filter
+        results, page = self._get_search_result(
+            url=self.url,
+            client=client,
+            search_params=search_params
+        )
+        assert len(results) >= 2  # there are > 1 results without the filter
+
+        # checking if `std:confval` filter is present
+        content = page.find('.navigable .language-list').text()
```

Comment:
Also we should fix the HTML, `language-list` isn't the class we should be using, it should be `role-list`.

Reply:
I also worry about depending on the HTML structure. This will break all the tests when we do our site redesign. Can we not capture this data from the query result itself? I believe the Django test client gives us template data directly, which we could use: https://docs.djangoproject.com/en/2.2/topics/testing/tools/#django.test.Response.context